### ISSUE TITLE
Task dependency does not work unless tasks are sorted in editor

### ISSUE THREAD

**Issue by shomodj** [mark-when/markwhen#106]
First, thank you! Its almost perfect.

I just have problem with task dependencies

Example:

```
after !task1 2 weeks: task 2
now - 2 weeks: task 1 !task1
```

It's really hard to create a complex project with dependencies if you have to order them in editor

Thank you

**Comment by kochrt** [comment]
Yeah I get that. Unfortunately how it works right now is the parser has to have seen the event id that is being referred to, because if it has seen it, then it is defined. 

Otherwise the parser would, after parsing the document, have to go back and create a graph of linked event ids. Probably doable, but you could potentially end up in cycles that don't make sense.

Like you could create something like this:
```
after !task1 1 week: !task0
after !task0 1 week: !task2
after !task2 1 week: !task1
```
but this group of events has no actual start or end date, they all just refer to each other.

For your use case I normally recommend ordering by dependencies when writing, then sorting in the timeline view, but you're right that this doesn't account for dependencies across groups 😞 .
